### PR TITLE
Optimization work

### PR DESCRIPTION
**Pull request by JohnTelman**
-The initial loading now takes about 3s to load (in local so need to add the time to download the textures on the live version)
-No more loading between tiles that would block the player
-The light system is run on a separate thread (light are added in the scene while you start playing)
-Textures are compressed to the maximum (further optimization can be done by an artist - the atlases should be edited because there is a lot of empty space)

### CODE REVIEW

File: packages/web/src/game/scenes/Game.ts [hunk 1]
```diff
@@ -243,125 +243,120 @@ export default class GameScene extends Scene {
...
-    // check in which map we're in
-    // west
-    if (playerPos.x - centerMapPos.x < -(level.pxWid / 2)) this._checkDir(level, 'w', false);
-    // east
-    else if (playerPos.x - centerMapPos.x > level.pxWid / 2) this._checkDir(level, 'e', false);
-    // north
-    if (playerPos.y - centerMapPos.y < -(level.pxHei / 2)) this._checkDir(level, 'n', false);
-    // south
-    else if (playerPos.y - centerMapPos.y > level.pxHei / 2) this._checkDir(level, 's', false);
+initializeGame(handshakeData: DataTypes[NetworkEvents.PLAYER_HANDSHAKE]) {
```
Comment (oscgu) [inline packages/web/src/game/scenes/Game.ts:323]: handshakeData is unused, do we need it?

File: packages/web/src/game/scenes/Game.ts [hunk 2]
```diff
@@ -548,18 +533,34 @@ export default class GameScene extends Scene {
...
+      const dayColorHex = this.dayColor.map(color => (Math.round(color * timeFactor)).toString(16)).join('');
+      const nightColorHex = this.nightColor.map(color => (Math.round(color * (0.2 - timeFactor))).toString(16)).join('');
+
+      // Determine which color to use based on time of day
+      let ambientColor;
+
+      if (Number.parseInt(dayColorHex, 16) < 272722)
+      {
+      // nighttime
+      ambientColor = Number.parseInt(nightColorHex, 16);
```
Comment (oscgu) [inline packages/web/src/game/scenes/Game.ts:554]: Please stay close to the code-style of the file (indentation, brackets etc.)

File: packages/web/src/game/world/LDtkParser.ts [hunk 3]
```diff
@@ -61,13 +61,19 @@ export class LdtkReader {
...
+    const tempScene = this.scene;
+
     let map: Phaser.Tilemaps.Tilemap;
     let tilesetObj = this.tilesets.find((t: any) => t.uid === layer.__tilesetDefUid)!;
     let tilesetWidth = tilesetObj.__cWid;
     let tileSize = layer.__gridSize;
 
-    map = this.scene.add.tilemap(layer.__identifier, tileSize, tileSize, layer.__cWid, layer.__cHei);
+
+    //map = this.scene.add.tilemap(layer.__identifier, tileSize, tileSize, layer.__cWid, layer.__cHei);
```
Comment (oscgu) [inline packages/web/src/game/world/LDtkParser.ts:75]: Remove if we dont need it

File: packages/web/src/game/entities/player/PlayerController.ts [hunk 4]
```diff
@@ -35,6 +35,8 @@ export default class PlayerController {
   update() {
     // if (Phaser.Input.Keyboard.JustUp((this.mainKeys as any).inventory)) this.player.toggleInventory();
 
+    //console.log("o");
```
Comment (oscgu) [inline packages/web/src/game/entities/player/PlayerController.ts:38]: Remove if we dont need it

**Review by oscgu — CHANGES_REQUESTED** [review]

File: packages/web/src/game/scenes/Game.ts [hunk 5]
```diff
@@ -199,12 +189,22 @@ export default class GameScene extends Scene {
...
+                      'NY_BushWick_BorderSouth', 'NY_Brownsville_Pawn', 'NY_BrookH_Hospital', 'NY_Brownsville_Depot'];
+
         // create map and entities
         this._mapHelper = new MapHelper(this);
-        this.mapHelper.createMap(data.current_map);
-        this.mapHelper.createEntities();
-        this.mapHelper.createCollisions();
+        let i : number = 0;
+
+        while (i < maps.length)
```
[the remark this answers is not in the document]
Comment (JohnTelman) [reply packages/web/src/game/scenes/Game.ts]: done

File: packages/web/src/game/scenes/Game.ts [hunk 6]
```diff
@@ -243,125 +243,120 @@ export default class GameScene extends Scene {
...
 
         this.initializeGame(data);
     });
 
     this.loadingSpinner = this.add.reactDom(loadingSpinner);
   }
 
-  update(time: number, delta: number) {
-    if (!this.initialized) return;
+  private initiatePlayers(data: any): void {
```
[the remark this answers is not in the document]
Comment (JohnTelman) [reply packages/web/src/game/scenes/Game.ts]: ok

File: packages/web/src/game/scenes/Game.ts [hunk 7]
```diff
@@ -243,125 +243,120 @@ export default class GameScene extends Scene {
...
+      hustler.setData('id', id);
+      hustler.currentMap = current_map;
+      if (current_map !== this.player.currentMap) {
+        hustler.setVisible(false);
+      }
+      this._hustlers.push(hustler);
+    }
+  }
+  
+  private initiateItemEntities(data: any): void {
```
[the remark this answers is not in the document]
Comment (JohnTelman) [reply packages/web/src/game/scenes/Game.ts]: ok

File: packages/web/src/game/scenes/Game.ts [hunk 8]
```diff
@@ -243,125 +243,120 @@ export default class GameScene extends Scene {
...
+
+private updateEntities(entities: Array<any>, playerCurrentMap: string) {
+  entities.forEach(entity => {
+    if (entity.currentMap === playerCurrentMap) {
+      entity.update();
+    }
+  });
+}
+
+update(time: number, delta: number) {
```
[the remark this answers is not in the document]
Comment (JohnTelman) [reply packages/web/src/game/scenes/Game.ts]: ok

File: packages/web/src/game/scenes/Game.ts (hunk 1 above)
Comment (JohnTelman) [reply packages/web/src/game/scenes/Game.ts]: ok

File: packages/web/src/game/scenes/Game.ts (hunk 2 above)
Comment (JohnTelman) [reply packages/web/src/game/scenes/Game.ts]: my bad

File: packages/web/src/game/world/LDtkParser.ts (hunk 3 above)
Comment (JohnTelman) [reply packages/web/src/game/world/LDtkParser.ts]: ok

File: packages/web/src/game/entities/player/PlayerController.ts (hunk 4 above)
Comment (JohnTelman) [reply packages/web/src/game/entities/player/PlayerController.ts]: ok

**Pull request merged**